### PR TITLE
Generic visitor.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by christianparpart — COMMENTED** [review]
I welcome the move from namespace `dev::solidity::assembly` into `dev`, as well as capability of not hard-failing when not all types were specified.

I am not sure about the name `GenericVisitor` though. Since all specialized visitors usually also carry the name, such as `ASTVisitor` and `ASTConstVisitor`, or `Scope::Visitor`, I'd suggest the name `Visitor`. (that's just my opinion though).

I am also not very sure if the suffix `FB` is intuitive to understand when reviewing code using that construct.

**Comment by chriseth** [comment]
I would also like to get some suggestions for the name, especially for the `FB` one.

File: libdevcore/Visitor.h [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+/// Generic visitor with fallback:
+/// boost::apply_visitor(GenericVisitorFB<Class1, Class2>(
+///     [](Class1& _c) { _c.f(); },
+///     [](Class2& _c) { _c.g(); }
+/// ), variant);
+/// This one DOES have a fallback and will NOT fail at
+/// compile-time if you do not specify all variants.
+
+template <class...>
+struct GenericVisitorFB{};
```
Comment (axic) [inline libdevcore/Visitor.h]: Why not `GenericFallbackVisitor`? Or something without an ambiguous `FB` shorthand?

Comment (christianparpart) [reply libdevcore/Visitor.h]: :+1:

**Comment by chriseth** [comment]
Renamed.

**Review by christianparpart — APPROVED** [review]
LGTM

**Pull request merged**